pandas-stubs/core/groupby/generic.pyi

```diff
@@ -61,7 +67,19 @@ class SeriesGroupBy(GroupBy, Generic[S1]):
     def agg(self, func: list[AggFuncTypeBase], *args, **kwargs) -> DataFrame: ...
     @overload
     def agg(self, func: AggFuncTypeBase, *args, **kwargs) -> Series: ...
-    def transform(self, func: Callable | str, *args, **kwargs) -> Series: ...
+    @overload
+    def transform(
+        self,
+        func: Callable[Concatenate[Series[S1], P], Series[S2]],
+        *args: Any,  # Ideally we would use P.args here, but it does not work with engine/engine_kwargs present
+        engine: Literal["cython", "numba", None] = ...,
+        engine_kwargs: dict[str, Any] | None = ...,
+        **kwargs: Any,  # Ideally we would use P.kwargs here, but does not work with engine/engine_kwargs present
```

Comment:
This was working great when I did not have `engine` and `engine_kwargs` in there because I could use `P.args` and `P.kwargs` as the annotation. However, I'm having trouble getting that working with these additional keyword arguments in there. Any ideas?

Reply:
Or maybe place `engine` and `engine_kwargs` before `*args` and `**kwargs` ???

Reply:
> Or maybe place `engine` and `engine_kwargs` before `*args` and `**kwargs` ???

So pyright no longer complains about that, but then the test doesn't work, because it thinks `engine` and `engine_kwargs` are positional parameters.

Still investigating...

Reply:
Looks like it isn't possible. Here is a similar issue: https://github.com/microsoft/pyright/issues/4189

The relevant part of PEP 612:

> Inside of `bar`, we would get `TypeError: foo() got multiple values for argument 'x'`. Requiring these concatenated arguments to be addressed positionally avoids this kind of problem, and simplifies the syntax for spelling these types. Note that this also why we have to reject signatures of the form `(*args: P.args, s: str, **kwargs: P.kwargs)` (See [Concatenating Keyword Parameters](https://peps.python.org/pep-0612/#concatenating-keyword-parameters) for more details).

This is unfortunate. I'll have to revert back to using `Any`.